### ISSUE TITLE
Content-Type is returned as "null" for extension-less files

### ISSUE THREAD

**Issue by clshortfuse** [webpack/webpack-dev-server#1614]
* Operating System: Mac OS X
* Node Version: v8.11.3
* NPM Version: 6.5.0
* webpack Version: 4.18.0
* webpack-dev-server Version: 3.18

- [ ] This is a **bug**
- [X] This is a **modification** request

### webpack.config.js

````
return {
    entry: entries,
    devtool: isProduction ? 'source-map' : 'nosources-source-map',
    mode: process.env.NODE_ENV || 'development',
    devServer: {
      host: '0.0.0.0',
      hot: false,
      inline: false,
    },
    output: {
      filename: '[name].min.js',
      path: path.resolve(__dirname, DEST, folder),
    },
    module: {
      rules: [{
        test: /\.js$/,
        include: [
          path.resolve(__dirname, 's3', folder),
        ],
        use: {
          loader: 'babel-loader',
          options: {
            presets: [
              ['@babel/preset-env'],
            ],
          },
        },
      }, {
        test: /\.scss$/,
        use: [
          'file-loader?name=[name].min.css',
          'extract-loader',
          'css-loader?import=false',
          {
            loader: 'postcss-loader',
            options: {
              plugins: () => [
                cssnano(),
              ],
            },
          },
          'sass-loader',
        ],
      }, {
        test: /_.+\.pug$/,
        use: [
          'html-loader',
          'pug-html-loader',
        ],
      }, {
        test: /index\.pug$/,
        use: [
          'file-loader?name=index.html',
          'extract-loader',
          'html-loader',
          'pug-html-loader',
        ],
      },
      {
        test: /\.pug$/,
        exclude: /(_.+)|(index)\.pug$/,
        use: [
          'file-loader?name=[name]',
          'extract-loader',
          'html-loader',
          'pug-html-loader',
        ],
      }],
    },
````

### Expected Behavior

Unknown mime/types should be able to have a default.

### Actual Behavior

`Content-Type` is reported as `null` for unknown types. 

### For Bugs; How can we reproduce the behavior?

Try hosting any file without an extension. `Content-Type` will be reported as `null`.

### For Features; What is the motivation and/or use-case for the feature?

Pages that don't have extensions might need a custom mime/type.

For example, one of my projects has the following output folder

* index.pug => /index.html
* login.pug => /login
* bundle.scss => /css/bundle.min.css
* bundle.js => /js/bundle.min.js

Chrome, Firefox, and Safari will treat anything with `Content-Type: null` as a `html/text` type. That's fine and it works as expected. IE11 and Edge will treat it, essentially, as `application/octet-stream`, which will trigger a download prompt.

It seems adding support for `mimeTypes` in `options` for `webpack-dev-middleware` should be a suitable fix.

**Comment by clshortfuse** [comment]
I "solved" my issue using the dirtiest of hacks. It appears `webpack-dev-middleware` sends a or rewrites the Content-Type and then uses a `res.send()` or `res.end()` command. It means it has final say what is sent so no other middleware gets a chance. By the time it gets to `options.after`, it's too late. 

`webpack-dev-middleware` doesn't null-check what is returned from `mime.getType()`, so it plugs in the adds a string to a null values. Thus, you get `null; charset=UTF-8`. 

So what I had to do is the following

````
mime.getType2 = mime.getType;
mime.getType = (key) => {
  if (key.indexOf('.') === -1) {
    return '';
  }
  return mime.getType2(key);
};
````

Clearly, this is unsafe, but you'll get `Content-Type: ; charset=UTF-8` which lets IE11 and Edge guess it's a `text/html` and open properly. I'll be closing this now and opening the issue with `webpack-dev-middleware`, since it seems that would be the more appropriate location.

**Comment by dmohns** [comment]
Hey, @clshortfuse Sorry to necro this. I am stuck in a similar scenario where `webpack-dev-middleware` sets an inappropriate `Content-Type` and I have no option to modify it in `after` for reason given in OP.
I am trying to make your monkey patch work by putting it into my `webpack.config.js`, but `webpack-dev-middleware` doesn't quite seem to pick it up. Where do I have to put the monkey patch?

**Comment by clshortfuse** [comment]
@dmohns Sorry for not getting back to you sooner.

In my `webpack.config.js` I added 

````
const mime = require('mime');

....

getConfig() {
  const config = {};

  // Setup config entries, modules, etc

  mime.getType2 = mime.getType;
  mime.getType = (key) => {
    if (key.indexOf('.') === -1) {
      return '';
    }
    return mime.getType2(key);
  };
  return config;
````

Basically, you import `mime` and change `getType` before `webpack-dev-middleware` executes. So when when it runs `mime.getType`, it actually runs through your proxy. I no longer need this because I actually filed the issue and made a pull-request to fix my issue. It was merged in 3.5.0. 

I do recommend you file the issue with more details over at that repository though. Feel free to ping me if you do. I wouldn't mind taking a look at it.
